pkg/backup/compaction_job.go

```diff
@@ -99,6 +99,28 @@ func maybeStartCompactionJob(
 		user,
 	)
 
+	// A race condition can occur where a compaction job ends after we fetch the
+	// backup chain but before we open a transaction to write the record. As a
+	// result, the written record is based on a chain that did not include the
+	// newly completed compaction job. As a result, it is possible that the chosen
```

Comment:
uber nit: you use the phrase "as a result" twice :D

Reply:
Ugh I hate it when I do that, thanks for catching it